core/domain/suggestion_registry.py

```diff
@@ -2001,3 +2001,511 @@ def __init__(
+    rejected_translations_count: int
+    rejected_translation_word_count: int
+    first_contribution_date: datetime.date
+    last_contribution_date: datetime.date
+
+
+class TranslationSubmitterTotalContributionStatsFrontendDict(TypedDict):
+    """Dictionary representing the TranslationSubmitterTotalContributionStats
+    object for frontend.
+    """
+
+    language_code: str
+    contributor_id: str
+    topic_ids_with_translation_submissions: List[str]
+    recent_review_outcomes: List[str]
+    recent_performance: int
+    overall_accuracy: float
+    submitted_translations_count: int
+    submitted_translation_word_count: int
+    accepted_translations_count: int
+    accepted_translations_without_reviewer_edits_count: int
+    accepted_translation_word_count: int
+    rejected_translations_count: int
+    rejected_translation_word_count: int
+    first_contribution_date: str
+    last_contributed_in_days: int
+
+
+class TranslationSubmitterTotalContributionStats:
+    """Domain object for the TranslationSubmitterTotalContributionStatsModel."""
+
+    def __init__(
+        self,
+        language_code: str,
+        contributor_id: str,
+        topic_ids_with_translation_submissions: List[str],
+        recent_review_outcomes: List[str],
+        recent_performance: int,
+        overall_accuracy: float,
+        submitted_translations_count: int,
+        submitted_translation_word_count: int,
+        accepted_translations_count: int,
+        accepted_translations_without_reviewer_edits_count: int,
+        accepted_translation_word_count: int,
+        rejected_translations_count: int,
+        rejected_translation_word_count: int,
+        first_contribution_date: datetime.date,
+        last_contribution_date: datetime.date
+    ) -> None:
+        self.language_code = language_code
+        self.contributor_id = contributor_id
+        self.topic_ids_with_translation_submissions = (
+            topic_ids_with_translation_submissions)
+        self.recent_review_outcomes = recent_review_outcomes
+        self.recent_performance = recent_performance
+        self.overall_accuracy = overall_accuracy
+        self.submitted_translations_count = submitted_translations_count
+        self.submitted_translation_word_count = submitted_translation_word_count
+        self.accepted_translations_count = accepted_translations_count
+        self.accepted_translations_without_reviewer_edits_count = (
+            accepted_translations_without_reviewer_edits_count
+        )
+        self.accepted_translation_word_count = accepted_translation_word_count
+        self.rejected_translations_count = rejected_translations_count
+        self.rejected_translation_word_count = rejected_translation_word_count
+        self.first_contribution_date = first_contribution_date
+        self.last_contribution_date = last_contribution_date
+
+    def to_dict(self) -> TranslationSubmitterTotalContributionStatsDict:
```

Comment:
Do we even need the non-frontend `to_dict` methods? Where/when are we planning to use them other then in tests?

Reply:
I added these methods since we might require them while editing services that updates the models or in beam jobs maybe. Not sure tho.

Reply:
I don't really foresee us needing the `to_dict` methods in the backend, since we should be operating on the domain objects in the backend. I would just remove them. Worst case you can always add them back later, but otherwise it will be dead code.